src/mono/mono/metadata/assembly.c

```diff
@@ -3156,12 +3156,42 @@ mono_assembly_get_name_internal (MonoAssembly *assembly)
 	return &assembly->aname;
 }
 
+/**
+ * mono_create_new_bundled_assembly:
+ */
+MonoBundledAssembly *
+mono_create_new_bundled_assembly (const char *name, const unsigned char *data, unsigned int size)
+{
+	MonoBundledAssembly *assembly = g_new0 (MonoBundledAssembly, 1);
+	assembly->name = strdup (name);
+	g_assert (assembly->name);
+	assembly->data = data;
+	assembly->size = size;
+	return assembly;
+}
+
+static void
+mono_free_bundled_assembly (const MonoBundledAssembly *assembly)
+{
+	g_free ((void *) assembly->name);
+	assembly->name = NULL;
+}
+
 /**
  * mono_register_bundled_assemblies:
  */
 void
 mono_register_bundled_assemblies (const MonoBundledAssembly **assemblies)
 {
+	if (bundles)
+	{
```

Comment:
Probably same ownership issues as with mono_register_bundle_assemblies, this is an existing API where caller owns memory.